libmamba/src/download/mirror_impl.cpp

```diff
@@ -250,9 +280,13 @@ namespace mamba::download
     {
         const AuthenticationData* data = get_authentication_data(split_path);
         std::string url = get_blob_url(split_path, data->sha256sum);
-        return MirrorRequest(initial_request, url);
+        std::vector<std::string> headers = { get_authentication_header(data->token) };
+
+        return MirrorRequest(initial_request, url, std::move(headers));
     }
 
+    // This is not used but could be if we use creds
+    // cf. OCIMirror constructor comment in header
     bool OCIMirror::need_authentication() const
```

Comment:
Removing `need_authentication()` from where it was used to avoid confusion as it's not clear how the credentials are supposed to be used. To pull artifacts we request a token, and we don't support uploading like `powerloader`, so AFAIK there is no need for credentials at the moment.
I left the function implementation though even if it's not used (as a hint for maybe later if needed, like `username` and `password`) but we could also decide this is useless and just remove it/them.